src/main/java/org/mockito/internal/matchers/text/ArrayIterator.java

```diff
@@ -33,6 +34,9 @@ public boolean hasNext() {
     }
 
     public Object next() {
+        if (!hasNext()) {
```

Comment:
Nice, it throws much better exception now when we forgot to check for next element.

It would be cool to add unit tests for this code :)